docs/apache-airflow-providers-apache-cassandra/index.rst

```diff
@@ -33,6 +33,7 @@ Content
     :caption: References
 
     Python API <_api/airflow/providers/apache/cassandra/index>
+    System Tests <_api/tests/system/providers/apache/cassandra/index>
```

Comment:
Why are we linking to this from user facing docs? That seems to me like a very odd thing to do.

In other words: I can't think of any reason why a user would ever visit and git anything useful from this page.

Reply:
Yes I also thought we might not need that. It's Mainly there add it to any toc tree- if we don't do it, sphinx complains that the index is missing from the toc tree. Do you know how to exclude those links in general? I am happy to remove them, but I simply have no idea how. Likely some exclude somewhere in sphinx autoapI. 

Previously example dags were excluded by `*/example_dags/*` and that's why it was not needed but witt the latest sphinx update - but we cannot exclude them this way because then they will be removed from the generated html (that was precisely the root cause of missing links previously - the "missing index in toc tree" was not generated but also link to example dags led to 404 because the generated example_dags pages were missing.

So if we know a better way to keep the files but not to have them included in any toc tree - that would be awesome. But I do no know how to do it.

Reply:
Ok I think I found a way.

Reply:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html?highlight=hidden#directive-toctree maybe?

Reply:
FYI. @ashb  The main problem was that :hidden: in toc tree did not remove the sidebar link. And I found out in the theme we can disable sidebar links for hidden TOC, but we were already using this in "apache-airfow" and "chart" (but luckily only there) to show sidebar content while not showing TOC in the page. 

Now we have hidden links shown in the sidebar in Airflow:

![image](https://user-images.githubusercontent.com/595491/173365633-0a5f2d75-57ce-4b3b-9b71-04d4728340a3.png)

And no  "system tests" links in provider's - neither sidebar, nor main page:

![image](https://user-images.githubusercontent.com/595491/173365865-ab20c8e7-f9b8-4f6c-83df-f76200f1b085.png)

But the sources are there.

Both for legacy exmple dags:

![image](https://user-images.githubusercontent.com/595491/173366123-b0558d5a-04c3-402b-9236-ba2190f3e69c.png)

And for new system tests:

![image](https://user-images.githubusercontent.com/595491/173366357-e2051922-b309-43ed-baed-139d9d3a6432.png)

Also we have consistent "example dags" links 

Wasn't THAT straightforward :). But finally seems to be there.



I also found a few inconsistencies in provider indexes that I fixed.  The system tests for google providers were moved without sub-folders (which was wrong). 

The `__init__.py` works nicely for docs scripts too. While enabling relative imports in "docs" I managed to make it in the way that:

* no need to insert syspath before imports
* build_docs.py ./publish_docs.py still works with shebang as executables and you can run it either with local airflow venv or in breeze for repeatability
* isort behaviour is consistent no matter if you only modify docs/* python files or run pre-commit for --all-files
* we have no implicit package in docs.

I also double-checked that adding __init__.py in docs did not mess with the airflow package (which I was afraid that could be side effect:

```
⌂85% θ92° [jarek:~/code/airflow] [airflow-3.7] fix-source-links-in-examples(+15/-1)+ 41s ± unzip -t dist/apache_airflow-2.4.0.dev0-py3-none-any.whl| grep doc
    testing: airflow/contrib/operators/docker_swarm_operator.py   OK
    testing: airflow/hooks/docker_hook.py   OK
    testing: airflow/operators/docker_operator.py   OK
    testing: airflow/providers/docker/__init__.py   OK
    testing: airflow/providers/docker/decorators/__init__.py   OK
    testing: airflow/providers/docker/decorators/docker.py   OK
    testing: airflow/providers/docker/hooks/__init__.py   OK
    testing: airflow/providers/docker/hooks/docker.py   OK
    testing: airflow/providers/docker/operators/__init__.py   OK
    testing: airflow/providers/docker/operators/docker.py   OK
    testing: airflow/providers/docker/operators/docker_swarm.py   OK
    testing: airflow/utils/docs.py    OK
    testing: airflow/www/static/dist/redoc.standalone.js   OK
    testing: airflow/www/static/dist/redoc.standalone.js.LICENSE.txt   OK
    testing: airflow/www/static/dist/redoc.standalone.js.map   OK
    testing: airflow/www/templates/airflow/redoc.html   OK
    testing: airflow/www/templates/appbuilder/dag_docs.html   OK
```